tests/serial/mapping-scaled-consistent/helpers.cpp

```diff
@@ -106,4 +106,203 @@ void testQuadMappingScaledConsistent(const std::string configFile, const TestCon
+{
+  using Eigen::Vector2d;
+
+  // MeshOne
+  Vector2d coordOneA{0.0, 0.0};
+  Vector2d coordOneB{1.0, 0.0};
+  Vector2d coordOneC{1.0, 1.0};
+  Vector2d coordOneD{0.0, 1.0};
+  Vector2d coordOneExtra{0.5, 1.0};
+  double   valOneA     = 4.0;
+  double   valOneB     = 3.0;
+  double   valOneC     = 2.0;
+  double   valOneD     = 1.0;
+  double   valOneExtra = 1.0;
+
+  // MeshTwo
+  Vector2d coordTwoA{0.0, 0.0};
+  Vector2d coordTwoB{1.0, 0.0};
+  Vector2d coordTwoC{1.0, 1.0};
+  Vector2d coordTwoD{0.0, 1.0};
+
+  double expectedIntegral = 7.0 / 3;
+
+  if (context.isNamed("SolverOne")) {
+    precice::SolverInterface interface("SolverOne", configFile, 0, 1);
+    // namespace is required because we are outside the fixture
+    const int meshOneID = interface.getMeshID("MeshOne");
+
+    // Setup mesh one.
+    int idA     = interface.setMeshVertex(meshOneID, coordOneA.data());
+    int idB     = interface.setMeshVertex(meshOneID, coordOneB.data());
+    int idC     = interface.setMeshVertex(meshOneID, coordOneC.data());
+    int idD     = interface.setMeshVertex(meshOneID, coordOneD.data());
+    int idExtra = interface.setMeshVertex(meshOneID, coordOneExtra.data());
+
+    interface.setMeshTriangleWithEdges(meshOneID, idA, idB, idExtra);
+    interface.setMeshTriangleWithEdges(meshOneID, idA, idD, idExtra);
+    interface.setMeshTriangleWithEdges(meshOneID, idB, idC, idExtra);
+
+    auto &mesh = testing::WhiteboxAccessor::impl(interface).mesh("MeshOne");
+    BOOST_REQUIRE(mesh.vertices().size() == 5);
+    BOOST_REQUIRE(mesh.triangles().size() == 3);
+
+    // Initialize, thus sending the mesh.
+    double maxDt = interface.initialize();
+    BOOST_TEST(interface.isCouplingOngoing(), "Sending participant should have to advance once!");
+
+    // Write the data to be send.
+    int dataAID = interface.getDataID("DataOne", meshOneID);
+    interface.writeScalarData(dataAID, idA, valOneA);
+    interface.writeScalarData(dataAID, idB, valOneB);
+    interface.writeScalarData(dataAID, idC, valOneC);
+    interface.writeScalarData(dataAID, idD, valOneD);
+    interface.writeScalarData(dataAID, idExtra, valOneExtra);
+
+    // Advance, thus send the data to the receiving partner.
+    interface.advance(maxDt);
+    BOOST_TEST(!interface.isCouplingOngoing(), "Sending participant should have to advance once!");
+    interface.finalize();
+  } else {
+    BOOST_TEST(context.isNamed("SolverTwo"));
+    precice::SolverInterface interface("SolverTwo", configFile, 0, 1);
+    // namespace is required because we are outside the fixture
+    int meshTwoID = interface.getMeshID("MeshTwo");
+
+    // Setup receiving mesh.
+    int idA = interface.setMeshVertex(meshTwoID, coordTwoA.data());
+    int idB = interface.setMeshVertex(meshTwoID, coordTwoB.data());
+    int idC = interface.setMeshVertex(meshTwoID, coordTwoC.data());
+    int idD = interface.setMeshVertex(meshTwoID, coordTwoD.data());
+
+    int idAB = interface.setMeshEdge(meshTwoID, idA, idB);
```

Comment:
Do we really need to store the return value here? CI claims that this value (and the following) is unused?

Reply:
How is that? The edge IDs are used to define the triangles. (Until we update to the Vertex-only API)

Reply:
Actually we can remove the edges completely since we use `SetMeshTriangleWithEdges`. Or should we rather convert to the vertex-only API?